src/main/scala/com/databricks/labs/overwatch/pipeline/PipelineFunctions.scala

```diff
@@ -165,6 +165,22 @@ object PipelineFunctions {
       .alias(defaultAlias)
   }
 
+  /**
+   * Builds the Map to be used in modifyStruct changeInventory to clean a new_cluster schema
+   * @param df df that contains the prefixed field to cleanse
+   * @param newClusterPrefix path to new_cluster including the "new_cluster" such as "new_settings.new_cluster"
+   * @return Map[String, Column] for changeInventory
+   */
+  def newClusterCleaner(df: DataFrame, newClusterPrefix: String): Map[String, Column] = {
+    Map(
+      s"${newClusterPrefix}.custom_tags" -> SchemaTools.structToMap(df, s"${newClusterPrefix}.custom_tags"),
+      s"${newClusterPrefix}.spark_conf" -> SchemaTools.structToMap(df, s"${newClusterPrefix}.spark_conf"),
+      s"${newClusterPrefix}.spark_env_vars" -> SchemaTools.structToMap(df, s"${newClusterPrefix}.spark_env_vars"),
+      s"${newClusterPrefix}.aws_attributes" -> SchemaTools.structToMap(df, s"${newClusterPrefix}.aws_attributes"),
+      s"${newClusterPrefix}.azure_attributes" -> SchemaTools.structToMap(df, s"${newClusterPrefix}.azure_attributes")
```

Comment:
It should stay as struct - it's controlled by us, so problems with changes by user won't happen

Reply:
if you're talking about cloud_attributes -- it's been converted to a map due to high cardinality in some cases. This causes the schema to be ever larger -- we're already hitting limits on the old default of 4000 chars in schema so all key:value pairs with higher cardinality of String:String have intentionally been converted to map several versions ago.